runtime/service/src/main/java/org/apache/polaris/service/events/jsonEventListener/aws/cloudwatch/AwsCloudWatchEventListener.java

```diff
@@ -0,0 +1,181 @@
+
+  private final String logGroup;
+  private final String logStream;
+  private final Region region;
+  private final boolean synchronousMode;
+  private final Clock clock;
+
+  @Inject CallContext callContext;
+
+  @Context SecurityContext securityContext;
+
+  @Inject
+  public AwsCloudWatchEventListener(AwsCloudWatchConfiguration config, Clock clock) {
+    this.logStream =
+        config
+            .awsCloudwatchlogStream()
+            .orElseThrow(
+                () -> new IllegalArgumentException("AWS CloudWatch log stream must be configured"));
+    this.logGroup =
+        config
+            .awsCloudwatchlogGroup()
+            .orElseThrow(
+                () -> new IllegalArgumentException("AWS CloudWatch log group must be configured"));
+    this.region =
+        Region.of(
+            config
+                .awsCloudwatchRegion()
+                .orElseThrow(
+                    () ->
+                        new IllegalArgumentException("AWS CloudWatch region must be configured")));
+    this.synchronousMode = config.synchronousMode();
+    this.clock = clock;
+  }
+
+  @PostConstruct
+  void start() {
+    this.client = createCloudWatchAsyncClient();
+    ensureLogGroupAndStream();
+  }
+
+  protected CloudWatchLogsAsyncClient createCloudWatchAsyncClient() {
+    return CloudWatchLogsAsyncClient.builder().region(region).build();
+  }
+
+  private void ensureLogGroupAndStream() {
+    try {
+      CompletableFuture<CreateLogGroupResponse> future =
+          client.createLogGroup(CreateLogGroupRequest.builder().logGroupName(logGroup).build());
+      future.join();
+    } catch (CompletionException e) {
+      if (e.getCause() instanceof ResourceAlreadyExistsException) {
+        LOGGER.debug("Log group {} already exists", logGroup);
+      } else {
+        throw e;
+      }
+    }
+
+    try {
+      CompletableFuture<CreateLogStreamResponse> future =
+          client.createLogStream(
+              CreateLogStreamRequest.builder()
+                  .logGroupName(logGroup)
+                  .logStreamName(logStream)
+                  .build());
+      future.join();
+    } catch (CompletionException e) {
+      if (e.getCause() instanceof ResourceAlreadyExistsException) {
+        LOGGER.debug("Log stream {} already exists", logStream);
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  @PreDestroy
+  void shutdown() {
+    if (client != null) {
+      client.close();
+    }
+  }
+
+  @Override
+  protected void transformAndSendEvent(HashMap<String, Object> properties) {
+    properties.put("realm", callContext.getRealmContext().getRealmIdentifier());
+    properties.put("principal", securityContext.getUserPrincipal().getName());
+    // TODO: Add request ID when it is available
+    String eventAsJson;
+    try {
+      eventAsJson = objectMapper.writeValueAsString(properties);
+    } catch (JsonProcessingException e) {
+      LOGGER.error("Error processing event into JSON string: {}", e.getMessage());
```

Comment:
Why not use 

Logger.error(String str, Throwable err)?

Reply:
Good call, added in next revision